ui/apps/platform/src/Containers/Collections/types.ts

```diff
@@ -1,46 +1,76 @@
 
-export type SelectorField =
-    | `${SelectorEntityType}`
-    | `${SelectorEntityType} Label`
-    | `${SelectorEntityType} Annotation`;
-
-type BaseSelectorRule = {
-    fieldName: SelectorField;
-    values: { value: string }[];
-};
+export type ByNameSelectorField = `${SelectorEntityType}`;
+export type ByLabelSelectorField = `${SelectorEntityType} Label`;
+export type ByAnnotationSelectorField = `${SelectorEntityType} Annotation`;
+
+export type SelectorField = ByNameSelectorField | ByLabelSelectorField | ByAnnotationSelectorField;
+
+const byNameRegExp = new RegExp(`^(${selectorEntityTypes.join('|')})$`);
+const byLabelRegExp = new RegExp(`^(${selectorEntityTypes.join('|')}) Label$`);
+const byAnnotationRegExp = new RegExp(`^(${selectorEntityTypes.join('|')}) Annotation$`);
+
+export function isByNameField(field: SelectorField): field is ByNameSelectorField {
+    return byNameRegExp.test(field);
+}
+
+export function isByLabelField(field: SelectorField): field is ByLabelSelectorField {
+    return byLabelRegExp.test(field);
+}
+
+export function isByAnnotationField(field: SelectorField): field is ByAnnotationSelectorField {
+    return byAnnotationRegExp.test(field);
+}
 
-export type DisjunctionSelectorRule = BaseSelectorRule & { operator: 'OR' };
-export type ConjunctionSelectorRule = BaseSelectorRule & { operator: 'AND' };
 /**
- * A valid `SelectorRule` can use either 'AND' or 'OR' operations to resolve values, but
- * since the current UI implementation only supports 'OR', we need to maintain separate types here.
+ * A valid server side `SelectorRule` can use either 'AND' or 'OR' operations to resolve values, but
+ * the current UI implementation only supports 'OR'.
  */
-export type SelectorRule = DisjunctionSelectorRule | ConjunctionSelectorRule;
+export type NameSelectorRule = {
+    operator: 'OR';
+    values: string[];
+};
 
-export type ResourceSelector = {
-    rules: SelectorRule[];
+export type LabelSelectorRule = {
+    operator: 'OR';
+    key: string;
+    values: string[];
 };
 
 /**
  * The front end currently only supports rules defined for names and labels, annotations are excluded.
  */
-export type SupportedSelectorField = Exclude<SelectorField, `${SelectorEntityType} Annotation`>;
+export type SupportedSelectorField = ByNameSelectorField | ByLabelSelectorField;
 
 export function isSupportedSelectorField(field: SelectorField): field is SupportedSelectorField {
-    return !field.endsWith('Annotation');
+    return isByNameField(field) || isByLabelField(field);
 }
 
-/**
- * This type extracts the `fieldName` property from the individual rules and groups them
- * using a single `field` property due to the UI only supporting a single field per entity
- * type in collection rules.
- */
-export type ScopedResourceSelector = {
-    field: SupportedSelectorField;
-    rules: Omit<DisjunctionSelectorRule, 'fieldName'>[];
-} | null;
+export type ByNameResourceSelector = {
```

Comment:
Note the difference between `rule` and `rules` in these two types. The UX MVP only allows a single "rule" when selecting by name (i.e. the entity matches any of a list of values), while the MVP supports multiple rules when selecting by label (i.e. the entity matches _all_ of a list of label key/values, with unique label keys being grouped together and matching on _any_ of their respective values.)